lib/link_expansion/hashed_edition.rb

```diff
@@ -0,0 +1,37 @@
+module LinkExpansion::HashedEdition
+  def edition_hash(values)
+    return nil unless values.present?
+    hash = values.is_a?(Array) ? hash_for(values) : values
+    hash = SymbolizeJSON.symbolize(hash)
+    hash = hash.slice(*edition_fields)
+    hash[:api_path] = api_path(hash) unless hash[:base_path].nil?
+    hash[:withdrawn] = withdrawn?(hash)
+    hash.delete(:id)
+    hash
+  end
+
+  def edition_fields
+    fields = LinkExpansion::Rules::DEFAULT_FIELDS_WITH_DETAILS.dup << :id << :state
+    fields -= %i[api_path withdrawn]
+    fields
+  end
+
+private
+
+  def hash_for(values)
+    return nil unless values.present?
+    Hash[edition_fields.zip(values)]
+  end
+
+
+  def api_path(attrs)
+    "/api/content" + attrs[:base_path]
+  end
+
+  def withdrawn?(attrs)
```

Comment:
In the case of a large taxonomy tree doing the LEFT OUTER JOIN onto unpublishings appears to be slower by a couple of hundred ms but I guess taxons can't be unpublished/withdrawn.
```
#<Benchmark::Tms:0x007fb235e553f0 @cstime=0.0, @cutime=0.0, @label="", @real=4.603115538000566, @stime=0.7899999999999998, @total=2.88, @utime=2.09>
```